demo/index.js

```diff
@@ -1,77 +1,306 @@
+            event.stopPropagation();
+
+            block.classList.remove("drag-over");
+
+            if (currentlyDraggedTask) {
+                const ul = block.querySelector("ul");
+                const tasks = block.querySelectorAll(".task");
+        
+                // Get mouse position
+                const mouseY = event.clientY;
+        
+                // Loop through tasks and find the correct position
+                let inserted = false;
+                for (let i = 0; i < tasks.length; i++) {
+                    const task = tasks[i];
+                    
+                    // Get the task's bounding rectangle
+                    const taskRect = task.getBoundingClientRect();
+                    
+                    // Calculate the midpoint of the task
+                    const taskMidY = taskRect.top + taskRect.height / 2;
+                    
+                    // If mouse is above the midpoint, insert before
+                    if (mouseY < taskMidY) {
+                        ul.insertBefore(currentlyDraggedTask, task);
+                        inserted = true;
+                        break;
+                    }
+                }
+        
+                // If not inserted above any task, append at the end
+                if (!inserted) {
+                    ul.appendChild(currentlyDraggedTask);
+                }
+            }
+        });
+    }
+}
+
+enableDragForKanbanBoard();
+
+/** Change the current year in the copyright. */
+const currentYear = new Date().getFullYear();
+const copyrights = document.querySelectorAll(`.current-year`);
+for (let i = 0; i < copyrights.length; i++) {
+    copyrights.item(i).innerHTML = currentYear;
+}
+
+/** Handle opening and closing the menu on mobile. */
+const navigationContainer = document.querySelector(`#navigation-menu`);
+const menuButton = document.querySelector(`#menu-button`);
+const backgroundOverlay = document.querySelector(`#background-overlay`);
+const closeButton = document.querySelector(`#close-button`);
+
+menuButton.addEventListener("click", () => {
+    navigationContainer.classList.toggle("open");
+});
+
+backgroundOverlay.addEventListener("click", () => {
+    navigationContainer.classList.remove("open");
+});
+
+closeButton.addEventListener("click", () => {
+    navigationContainer.classList.remove("open");
+});
+
+/** Close the menu when a link is clicked. */
+const menuLinks = navigationContainer.getElementsByTagName("a");
+for (let i = 0; i < menuLinks.length; i++) {
+    menuLinks.item(i).addEventListener("click", () => {
+        navigationContainer.classList.remove("open");
+    });
+}
+
+/** Highlight the code with Shiki. */
+const downloadPolyfillCode = document.querySelector(`#download-polyfill-code`);
+downloadPolyfillCode.innerHTML = await codeToHtml(
+    `<script
+    src="drag-drop-touch.esm.min.js?autoload"
+    type="module"
+></script>`,
+    {
+        lang: "javascript",
+        theme: "catppuccin-latte",
+    }
+);
```

Comment:
If we're inlining SVG, let's also inline the syntax-highlighted code. That way, this code only needs to run once (at build time) rather than having to run as many times as there are people visiting this page.